contracts/exchange/Exchange.sol

```diff
@@ -65,6 +65,55 @@ contract Exchange is IExchange, ContextUpgradeable, OwnableUpgradeable, ERC165St
         emit OrderPlaced(_msgSender(), id, _order);
     }
 
+    function fillOrder(
+        uint256 _orderId,
+        uint256 amountToFill,
+        uint256 maxSpend
+    ) external override {
+        // convert uint256 orderId to array for use in functions
+        uint256[] memory orderIdArray = new uint256[](1);
+        orderIdArray[0] = _orderId;
+        // Verify orders exist
+        require(orderbook.verifyOrdersExist(orderIdArray), "Non-existent order");
```

Comment:
couldn't you make single version of these functions as well (verifyOrdersExist(), getOrderAmounts(), getPaymentTotals(), fillOrders())? Can you check if that saves gas too.